django/db/models/expressions.py

```diff
@@ -513,6 +514,32 @@ def copy(self):
         copy.extra = self.extra.copy()
         return copy
 
+    @classmethod
+    def as_transform(cls, lookup_name, bilateral=False, output_field=None, expressions=None):
+        def as_sql(self, compiler, connection):
```

Comment:
How about moving this factory to `Transform.from_expression(…)` and having this method call that? It does not seem like it's meant to be overloaded in subclasses and it would keep the layers separate without the `Func` object having to know how `Transforms` work.